alpaca_trade_api/rest_async.py

```diff
@@ -0,0 +1,183 @@
+import os
+import aiohttp
+import asyncio
+from alpaca_trade_api.entity_v2 import BarsV2, QuotesV2, TradesV2, \
+    EntityList, TradeV2, QuoteV2
+import pandas as pd
+from alpaca_trade_api.common import URL, get_api_version, get_data_url
+
+NY = 'America/New_York'
+
+
+class AsyncRest:
+    def __init__(self,
+                 key_id: str = None,
+                 secret_key: str = None,
+                 data_url: URL = None,
+                 api_version: str = None,
+                 oauth=None,
+                 raw_data: bool = False
+                 ):
+        """
+        :param raw_data: should we return api response raw or wrap it with
+                         Entity objects.
+        """
+        self._key_id, self._secret_key = key_id, secret_key
+        self._data_url: URL = URL(data_url or get_data_url())
+        self._api_version = get_api_version(api_version)
+        self._use_raw_data = raw_data
+        self._retry = int(os.environ.get('APCA_RETRY_MAX', 3))
+        self._retry_wait = int(os.environ.get('APCA_RETRY_WAIT', 3))
+
+    def _get_historic_url(self, _type, symbol):
+        return f"{self._data_url}/v2/stocks/{symbol}/{_type}"
+
+    def _get_latest_url(self, _type, symbol):
+        return f"{self._data_url}/v2/stocks/{symbol}/{_type}/latest"
+
+    async def _iterate_requests(self,
+                                symbol,
+                                payload,
+                                limit,
+                                entity_type: str,
+                                entity_list_type: EntityList) -> pd.DataFrame:
+        """
+        iterates the api asynchronously until we get all requested data
+        :param symbol:
+        :param payload:
+        :param entity_type: bars/trades/quotes
+        :param entity_list_type:
+        :return:
+        """
+        df = pd.DataFrame({})
+        url = self._get_historic_url(entity_type, symbol)
+        async for packet in self._request(symbol, url, payload):
+            if packet.get(entity_type):
+                response = entity_list_type(packet[entity_type]).df
+                df = pd.concat([df, response], axis=0)
+                if len(df) >= limit:
+                    break
+
+        return df
+
+    async def get_bars_async(self, symbol, start, end, timeframe, limit=1000):
+        _type = "bars"
+
+        payload = {
+            "adjustment": 'raw',
```

Comment:
[minor] This should be an argument instead

Reply:
added